modules/tinymce/src/core/main/ts/newline/InsertNewLine.ts

```diff
@@ -5,53 +5,56 @@ import { getNewlineBehavior } from '../api/Options';
 import { EditorEvent } from '../api/util/EventDispatcher';
 import { execDeleteCommand } from '../delete/DeleteUtils';
 import { fireFakeBeforeInputEvent, fireFakeInputEvent } from '../keyboard/FakeInputEvents';
-import * as InsertBlock from './InsertBlock';
-import * as InsertBr from './InsertBr';
+import { blockbreak } from './InsertBlock';
+import { linebreak } from './InsertBr';
 import * as NewLineAction from './NewLineAction';
 
-const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>) => {
-  const insertBreak = (breakType: (editor: Editor, evt?: EditorEvent<KeyboardEvent>) => void, fakeEvent: string) => () => {
-    if (!editor.selection.isCollapsed()) {
-      execDeleteCommand(editor);
-    }
-    if (Type.isNonNullable(evt)) {
-      const event = fireFakeBeforeInputEvent(editor, fakeEvent);
-      if (event.isDefaultPrevented()) {
-        return;
-      }
+const insertBreak = (breakType: {
```

Comment:
Should this be it's own type a BreakType i guess it would have to be a separate module since three modules uses that type. I just tend to want to use named types for things since it is easier when you refactor but maybe overkill here.

Reply:
We wouldn't need to make all 3 modules use the type, it would just be structurally equal as it is today using an anonymous type. I wasn't sure which style we prefer but thought I'd give this a go for now.